dd/dd.go

```diff
@@ -0,0 +1,84 @@
+// dd holds the interfaces and structures shared by Datadog's tracer packages.
+// It is a good overview of the available API and functionalities.
+package dd
```

Comment:
I've tried but if we do that it will result in a circular dependency:
- `internal` imports `tracer` for these interfaces
- `tracer` imports `internal` for the `GlobalTraces`
So we have to keep it.